logger/test_logger_integration.py

```diff
@@ -0,0 +1,53 @@
+"""
+Integration tests for the logger.
+"""
+
+import inspect
+
+from logger.modules import logger, logger_setup_main
+from read_yaml.modules import read_yaml
+
+
+def main() -> int:
```

Comment:
Can this not be tested with pytest? If all logger does is write to stdout and stderr, you should be able to test that using [pytest](https://docs.pytest.org/en/stable/how-to/capture-stdout-stderr.html). If so, I would recommend making this a pytest because at least I would be too lazy to manually run all these tests lmao. (I'm not too familiar with python's logging module)